### ISSUE TITLE
Lidarr - Unknown scan request from: '127.0.0.1'

### ISSUE THREAD

**Issue by dabigc** [l3uddz/plex_autoscan#89]
Lidarr was working with plex_autoscan but has stopped and gives this error in the plex_autoscan logs when Lidarr attempts to trigger a scan.

`ERROR -  AUTOSCAN [139816710383424]: Unknown scan request from: '127.0.0.1'`

It looks like the Lidarr web hook options have changed (as seen below) from what the [documentation](https://github.com/l3uddz/plex_autoscan#lidarr) shows. Is this is a known issue or is it something I can help get working?

<img width="722" alt="image" src="https://user-images.githubusercontent.com/1239774/57055792-48c5de80-6c63-11e9-8ca2-3295ee89b9e8.png">

**Comment by desimaniac** [comment]
Looks like they made some change on their side, needs updating.

**Comment by wishbones** [comment]
Here is a patch that fixes it:

```patch
diff --git a/scan.py b/scan.py
index aa6a08d..4b40cd6 100755
--- a/scan.py
+++ b/scan.py
@@ -514,14 +514,18 @@ def client_pushed():
         start_scan(final_path, 'Sonarr',
                    "Upgrade" if ('isUpgrade' in data and data['isUpgrade']) else data['eventType'])
 
-    elif 'artist' in data and 'trackFile' in data and 'eventType' in data:
+    elif 'artist' in data and 'trackFiles' in data and 'eventType' in data:
         # lidarr download/upgrade webhook
-        path = os.path.join(data['artist']['path'], data['trackFile']['relativePath'])
-        logger.info("Client %r scan request for album track: '%s', event: '%s'", request.remote_addr, path,
-                    "Upgrade" if ('isUpgrade' in data and data['isUpgrade']) else data['eventType'])
-        final_path = utils.map_pushed_path(conf.configs, path)
-        start_scan(final_path, 'Lidarr',
-                   "Upgrade" if ('isUpgrade' in data and data['isUpgrade']) else data['eventType'])
+        paths = set()
+        for track in data['trackFiles']:
+            path = os.path.join(data['artist']['path'], track['relativePath'])
+            paths.add(path)
+        for path in paths:
+            logger.info("Client %r scan request for album track: '%s', event: '%s'", request.remote_addr, path,
+                        "Upgrade" if ('isUpgrade' in data and data['isUpgrade']) else data['eventType'])
+            final_path = utils.map_pushed_path(conf.configs, path)
+            start_scan(final_path, 'Lidarr',
+                       "Upgrade" if ('isUpgrade' in data and data['isUpgrade']) else data['eventType'])
 
     else:
         logger.error("Unknown scan request from: %r", request.remote_addr)
```

**Comment by NHDriver4** [comment]
Was there ever a pr generated to fix this issue? I'm also having unknown scan requests from Lidarr.

**Comment by desimaniac** [comment]
Fix with 94d7d3c
